pkg/gadgets/run/tracer/tracer.go

```diff
@@ -177,6 +177,51 @@ func (t *Tracer) Close() {
 	}
 }
 
+func ringbufAvailable() bool {
+	_, err := ebpf.NewMap(&ebpf.MapSpec{
+		Type:       ebpf.RingBuf,
+		MaxEntries: uint32(os.Getpagesize()),
+	})
+
+	return err == nil
+}
+
+func (t *Tracer) loadeBPFObjects(opts ebpf.CollectionOptions, bufName string) error {
+	available := ringbufAvailable()
+
+	if !available {
+		if bufName != "" {
+			bufMap, ok := t.spec.Maps[bufName]
+			if !ok {
+				return fmt.Errorf("no buffer map named %s", bufName)
+			}
+
+			bufMap.Type = ebpf.PerfEventArray
+			bufMap.KeySize = 4
+			bufMap.ValueSize = 4
+
+			t.spec.Maps[bufName] = bufMap
+		}
+	}
```

Comment:
Print debug messages indicating ring_buf is not available and that logic will fallback to perf event array.